doc/renaming_index.xml

```diff
@@ -2,3 +2,7 @@
 <Index Key="IsRecognitionInfo"><C>IsRecognitionInfo</C></Index>
 <Index Key="RIFac"><C>RIFac</C></Index>
 <Index Key="RIKer"><C>RIKer</C></Index>
+<Index Key="RIParent"><C>RIParent</C></Index>
+<Index Key="methodsforfactor"><C>methodsforfactor</C></Index>
+<Index Key="forfactor"><C>forfactor</C></Index>
+<Index Key="forkernel"><C>forkernel</C></Index>
```

Comment:
Not about this PR, but: why is there a separate index file? Why aren't those index entries directly in `renaming_table.xml`?

Reply:
I changed our script such that the index is also written into the `renaming_table.xml`.

Reply:
And I also tested whether the index still works. Everything is fine.